### PR TITLE
JSON serialization fix

### LINKED ISSUE #499: fix provider_data in send_invoice
## Failure Information (for bugs)

provider_data in send_invoice method should be encoded to JSON or should be str like it is in the docs, currently it is Optional[Dict]

### PR DESCRIPTION
**Pull request by evgfilim1**
# Description

* Replace `str` with [`aiogram.utils.payload.prepare_arg`](https://github.com/aiogram/aiogram/blob/f73f7874a3b4a11b5d906b3804a9bc923015ccc2/aiogram/utils/payload.py#L46) in [`aiogram.bot.api.compose_data`](https://github.com/aiogram/aiogram/blob/f73f7874a3b4a11b5d906b3804a9bc923015ccc2/aiogram/bot/api.py#L158).
* Remove `prepare_arg` from bot methods.

This is made to prevent some future and existing bugs with serialization, such as https://github.com/aiogram/aiogram/issues/499#issuecomment-828314687. Also I described the bug in aiogram dev chat in Telegram (message from 2021-04-28 06:21:51 UTC).

**MUST BE TESTED ON REAL BOTS**. While I'm not *really* sure it doesn't break anything, I'll put `breaking` label.

Fixes #499 

# How has this been tested?

- Run pytest on `tests` directory
- Made test request manually:
  ```python
  # imports omitted...
  async def main():
      await Bot('TOKEN').send_message(
          chat_id=123456789,
          text='test https://t.me/evgfilim1',
          reply_markup=InlineKeyboardMarkup().add(
              InlineKeyboardButton('as1', callback_data='df1'),
              InlineKeyboardButton('as2', callback_data='df2'),
              InlineKeyboardButton('as3', callback_data='df3'),
          ),
          disable_notification=True,
          disable_web_page_preview=True,
      )
  if __name__ == '__main__':
      asyncio.run(main())
  ```

**Test Configuration**:
* Operating System: ArchLinux with Linux 5.11.16-arch1
* Python version: 3.9.3
* aiogram version: based on commit 4120408aa314bd53306805da508bd292323c3706
* pip freeze: https://paste.aiogram.dev/hemeqicewe.apache

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation (not needed)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (no new tests needed I guess)
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by uwinx — APPROVED** [review]
looks good to me. great job!

File: aiogram/utils/payload.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def prepare_arg(value):
         return int((now + value).timestamp())
     if isinstance(value, datetime.datetime):
         return round(value.timestamp())
-    if isinstance(value, LazyProxy):
+    if isinstance(value, (LazyProxy, int, float)):
```
Comment (uwinx) [inline aiogram/utils/payload.py:62]: we can add support for `decimal.Decimal` in here.

Comment (evgfilim1) [reply aiogram/utils/payload.py:62]: Its support is not declared anywhere, so I think we don't need to add it :thinking:

Comment (JrooTJunior) [reply aiogram/utils/payload.py:62]: Why do we need to convert float and int to str?

**Review by JrooTJunior — COMMENTED** [review]
With this change too much prepare_arg calls will be made. Don't like it. It make too much `isinstance` calls is not good thing for performance

File: aiogram/utils/payload.py (hunk 1 above)
Comment (evgfilim1) [reply aiogram/utils/payload.py:62]: @JrooTJunior this is the current behavior in 2.x — casting everything (that wasn't yet prepared by `prepare_arg`) in `compose_data` to `str` before sending:
https://github.com/aiogram/aiogram/blob/d5a4c0c4afc6f03b9de19f7f68f852c9188cf072/aiogram/bot/api.py#L169

**Comment by evgfilim1** [comment]
Closing this as I don't have an idea how to make performance better

**Pull request closed without merging**